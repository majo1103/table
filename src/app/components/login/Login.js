import React from "react";

import { Redirect } from "react-router-dom";

import { client } from "../../Client";
import "./Login-Form-Dark.css";

export class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      shouldRedirect: false,
      message: ""
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.performLogin = this.performLogin.bind(this);
    this.redirectPath = this.redirectPath.bind(this);
  }

  validateForm() {
    return this.state.email.length > 0 && this.state.password.length > 0;
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
  }

  performLogin(event) {
    if (client.login(this.state.email)) {
      this.setState({ shouldRedirect: true });
    } else {
      this.setState({ message: "Please register first" });
    }
  }

  redirectPath() {
    const locationState = this.props.location.state;
    const pathname =
      locationState && locationState.from && locationState.from.pathname;
    return pathname || "/poll";
  }

  render() {
    if (this.state.shouldRedirect) {
      return <Redirect to={this.redirectPath()} />;
    } else {
      return (
        <div className="login-dark">
          <form onSubmit={this.handleSubmit}>
            <div className="illustration">
              <i className="icon ion-ios-locked-outline" />
            </div>
            <div className="form-group">
              <input
                className="form-control"
                type="email"
                name="email"
                placeholder="Email"
                onChange={this.handleChange}
                autoFocus
              />
            </div>
            <div className="form-group">
              <input
                className="form-control"
                type="password"
                name="password"
                placeholder="Password"
                onChange={this.handleChange}
              />
            </div>
            <div className="form-group">
              <button
                className="btn btn-primary btn-block"
                type="submit"
                onClick={this.performLogin}
                disabled={!this.validateForm()}
              >
                Log In
              </button>
            </div>
            <div>
              <label name="message" className="forgot">
                {this.state.message}
              </label>
            </div>
          </form>
        </div>
      );
    }
  }
}
