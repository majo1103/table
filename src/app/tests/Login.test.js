import { Login } from "../components/login/Login";
import React from "react";
import { render } from "react-testing-library";

it("should have the `button` element", () => {
  const wrapper = render(<Login />);
  expect(wrapper).toMatchSnapshot();
});
