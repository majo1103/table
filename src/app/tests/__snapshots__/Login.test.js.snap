// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have the \`button\` element 1`] = `
Object {
  "baseElement": <body>
    <div>
      <div
        class="login-dark"
      >
        <form>
          <div
            class="illustration"
          >
            <i
              class="icon ion-ios-locked-outline"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="email"
              placeholder="Email"
              type="email"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="password"
              placeholder="Password"
              type="password"
            />
          </div>
          <div
            class="form-group"
          >
            <button
              class="btn btn-primary btn-block"
              disabled=""
              type="submit"
            >
              Log In
            </button>
          </div>
          <div>
            <label
              class="forgot"
              name="message"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-dark"
    >
      <form>
        <div
          class="illustration"
        >
          <i
            class="icon ion-ios-locked-outline"
          />
        </div>
        <div
          class="form-group"
        >
          <input
            class="form-control"
            name="email"
            placeholder="Email"
            type="email"
          />
        </div>
        <div
          class="form-group"
        >
          <input
            class="form-control"
            name="password"
            placeholder="Password"
            type="password"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-primary btn-block"
            disabled=""
            type="submit"
          >
            Log In
          </button>
        </div>
        <div>
          <label
            class="forgot"
            name="message"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
